platform/mbed_assert.h

```diff
@@ -49,6 +49,65 @@ do {                                                     \
 } while (0)
 #endif
 
+
+#define _MBED_CONCAT_(a, b) a##b
+#define _MBED_CONCAT(a, b) _MBED_CONCAT_(a, b)
+#define _MBED_ASSERTION _MBED_CONCAT(_MBED_ASSERTION_AT_, __LINE__)
+
+/** MBED_STATIC_ASSERT
+ *  Declare compile-time assertions, results in compile-time error if condition is false
+ *
+ *  The assertion acts as a declaration that can be placed at file scope, in a
```

Comment:
Is it possible to enhance the documentation by explaining that this macro is valid in C and C++ at file scope and in a code block. Then split the structure case in two: 
* C++: this macro is valid.
* C: should use `MBED_STRUCT_STATIC_ASSERT` which is also valid for C++ code.